org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLFormatter.java

```diff
@@ -51,12 +51,12 @@ public XMLFormatter(XMLExtensionsRegistry extensionsRegistry) {
 	 */
 	public List<? extends TextEdit> format(DOMDocument xmlDocument, Range range, SharedSettings sharedSettings) {
 		try {
-			if (sharedSettings.getFormattingSettings().isExperimental()) {
-				XMLFormatterDocumentNew formatterDocument = new XMLFormatterDocumentNew(xmlDocument, range,
+			if (sharedSettings.getFormattingSettings().isLegacy()) {
+				XMLFormatterDocument formatterDocument = new XMLFormatterDocument(xmlDocument.getTextDocument(), range,

```

Comment:
I think we should rename `XMLFormatterDocument` to `XMLFormatterDocumentOld` or something similar to avoid confusion. Make sure to update the javadoc at the top of the `XMLFormatterDocument` class.